docs/developer-guide/loading-data.md

```diff
@@ -0,0 +1,203 @@
+const deck = new Deck({...});
+
+function updateData() {
+  const layer = new ScatterplotLayer({
+    data: `path/to/data.json?t=${Date.now()}`
+  });
+
+  deck.setProps({layers: [layer]});
+};
+
+setInterval(updateData, 5 * 60 * 1000);
+```
+
+## Load Resource Without an URL
+
+In some use cases, resources do not exist at a static URL. For example, some applications construct images dynamically based on user input. Some applications receive arbitrary binary blobs from a server via a WebSocket connection.
+
+Before reading on, remember that you don't have to use a loader if your app already knows how to interpret the content. For example, if you have the RGBA values of all pixels of an image. you can simply construct an [ImageData](https://developer.mozilla.org/en-US/docs/Web/API/ImageData) object:
+
+```js
+new BitmapLayer({
+  image: new ImageData(pixels, 128, 128)
+})
+```
+
+If you have a custom-formatted binary, consider the techniques in [using binary data](/docs/developer-guide/performance.md#use-binary-data).
+
+The following examples only address the use cases where you need a loader/parser to interpret the incoming data.
+
+### Example: Use image from a programmatically generated SVG string
+
+The following code dynamically generates SVG icons and convert them to [data URLs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs).
+
+```js
+function createSVGIcon(number) {
+  const label = number < 10 ? number.toString : '10+';
+  return `\
+  <svg width="24" height="24" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
+    <circle cx="12" cy="12" r="10" fill="#c00" stroke="#fa1" stroke-width="2"/>
+    <text x="12" y="12" fill="#fff" text-anchor="middle" alignment-baseline="middle" font-size="8">${label}</text>
+  </svg>`;
+}
+
+// Note that a xml string cannot be directly embedded in a data URL
+// it has to be either escaped or converted to base64.
+function svgToDataURL(svg) {
+  return `data:image/svg+xml;charset=utf-8,${encodeURIComponent(svg)}`;
+  // or
+  return `data:image/svg+xml;base64,${btoa(svg)}`;
+}
+
+new IconLayer({
+  getIcon: d => {
+    icon: svgToDataURL(createSVGIcon(d.value)),
+    width: 24,
+    height: 24
+  }
+})
+```
+
+### Example: Parse glTF from a binary blob
+
+The following code shows how to parse a glTF model that is already loaded into an [ArrayBuffer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer) object.
+
+There are two ways for deck.gl to load it. One is to create a blob URL:
+
+```js
+const blob = new Blob([arraybuffer]);
+const objectURL = URL.createObjectURL(blob);
+
+new ScenegraphLayer({
+  scenegraph: objectURL
+});
+```
+
+Or more directly, import the [parse](https://loaders.gl/modules/core/docs/api-reference/parse) utility from loaders.gl (already a dependency of deck.gl), which returns a promise:
+
+```js
+import {parse} from '@loaders.gl/core';
+import {GLTFLoader} from '@loaders.gl/gltf';
+
+new ScenegraphLayer({
```

Comment:
Fine, but it feels kind of weird as loaders.gl is designed to always parses arrayBuffers and avoid need for createObjectURL and pushing things into browser blob storage.

Right now deck.gl only call loaders if data is a URL. If it was an arrayBuffer we could also do so. Perhaps activate via a `loadOption`?

Reply:
We can't. That will break the binary data use case.

Reply:
Doesn't that involve a `data` object? Does data actually already accept an ArrayBuffer?

Reply:
Yes, `data` can be an ArrayBuffer. Other async props follow the same logic.

Reply:
Understood. At some point it might be good to add something to the docs about that. 

At least, searching the deck.gl docs for `ArrayBuffer` turns up nothing (using the website search function) .

And the `data` prop is documented thus: https://deck.gl/docs/api-reference/core/layer#data

Reply:
ArrayBuffer is an iterable.

Reply:
ArrayBuffer is a non-iterable. `data` is basically allowed to be any object.